infra/stream/test/TestFormatter.cpp

```diff
@@ -0,0 +1,241 @@
+#include "infra/stream/Formatter.hpp"
+#include "gtest/gtest.h"
+
+class TestFormatObject
+{
+public:
+    int v{ 9 };
+    bool b{ true };
+};
+template<>
+void infra::Formatter<TestFormatObject &>::Format(infra::StreamWriter& writer, infra::FormatSpec& spec)
+{
+    Formatter<decltype(value.v)>(value.v).Format(writer, spec);
+}
+
+class FormatTest : public testing::Test
+{
+public:
+    FormatTest()
+        : writer(streamWriter.Writer())
+    {}
+protected:
+    void AssertFormatter(const char* expected) const
+    {
+        EXPECT_EQ(expected, streamWriter.Storage());
+    }
+    infra::StringOutputStream::WithStorage<60> streamWriter;
+    infra::StreamWriter& writer;
+};
+TEST_F(FormatTest, emptyString)
+{
+    Format(writer, "");
+    AssertFormatter("");
+}
+TEST_F(FormatTest, simpleString)
+{
+    Format(writer, "a");
+    AssertFormatter("a");
+}
+TEST_F(FormatTest, missingArgument)
+{
+    Format(writer, "{}");
+    AssertFormatter("}");
```

Comment:
Why do we expect to see "}" here? Where is this semantic derived from? Same goes for other error cases below.